Deadlock.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CodeReviewDemo
+{
```

Comment:
Consider using more descriptive variable names for the locks, such as `resourceLock1` and `resourceLock2`, to make the purpose of each lock clearer.